cdisc-org/conformance-rules-editor
# ISS-335: move rule schema from rules editor to core rules engine repo

Depends on: https://github.com/cdisc-org/cdisc-rules-engine/pull/409

To test before the rule engine PR is merged, you can add this env var to `local.settings.json`:
- `"RULE_SCHEMA_URL": "https://raw.githubusercontent.com/cdisc-org/cdisc-rules-engine/ISS-335-move-rule-schema-from-rules-editor-to-core-rules-engine-repo/resources/schema/CORE-base.json"`

To test after the merge:
- `"RULE_SCHEMA_URL": "https://raw.githubusercontent.com/cdisc-org/cdisc-rules-engine/main/resources/schema/CORE-base.json"`

Test steps:
1. Confirm the env variables are configured correctly in the 3 static web app envs (dev, training, prod) via the azure portal
1. Run the local rule editor using the appropriate env var from above
1. Login to the local rule editor
1. Navigate to http://localhost:4280/api/schema and confirm the json schema is returned
1. Navigate back to http://localhost:4280
1. Confirm the main page shows the rule template before selecting a rule
1. Select a rule
1. Ensure intellisense works in the editor pane
    - delete an operator name and allow the editor to suggest an operator with ctrl/cmd+space
1. Ensure validation works in the editor pane
    - provide an invalid operator name and the editor should show yellow lines
    - hover over the yellow lines and the editor should provide a tooltip with valid operator names. At the end of the tooltip, it should provide the correct location of `yaml-schema`
1. Ensure validation works in the testing pane
    - provide an invalid operator name
    - click on "Test" -> "Validate YAML against schema"
    - Confirm there is a red X and a list of validation errors in json
